### PR TITLE
Create CSS Entry for floats

### LINKED ISSUE #482: [Entry] CSS: Floats
### Entry Type (select all that apply)

New Entry for Existing Subject

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by ethan-master-coding**
### Description

Made the entry for CSS floats.

Closes #482 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/css/concepts/floats/floats.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+CatalogContent:
+  - 'learn-css'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The `float` propertie moves an element with in its parent.
+
+## Syntax
+
+The `float` property takes 5 different values. They are, `left`, `right`, and `none`. The value `left` makes the element float on the left of a parent. The value `right` makes the element float on the right of a parent. The value `none` makes the element not float at all.
```
Comment (pinalpithadiya) [inline content/css/concepts/floats/floats.md:18]: Here you have added 5 values and you have described only 3, could you please add other 2 as well i.e. initial and inherit? with description?

**Review by pinalpithadiya — COMMENTED** [review]
The file name can be changed to float from floats as this is a single css property having multiple assignable values.

**Comment by ethan-master-coding** [comment]
@pinalpithadiya Made your suggested changes.

**Comment by pinalpithadiya** [comment]
The checks are also failing for the PR here. you can check the issue in the failing check and fix it to make all these green :)

**Comment by ethan-master-coding** [comment]
@pinalpithadiya I know 2 checks are failing. I honestly don't know what each check is (I should learn what they are) and by the time all 3 reviews are done they are all passing anyways

**Comment by pinalpithadiya** [comment]
@ethan-master-coding  Yes, I can understand it is a minor thing, it is saying that the code blocks must have empty line before and after, which suggests, you should put an empty line before the css code block. :)

**Comment by ethan-master-coding** [comment]
@pinalpithadiya 

> it is saying that the code blocks must have an empty line before and after, which suggests, you should put an empty line before the CSS code block. :)

I put an empty line around the code blocks.

**Comment by pinalpithadiya** [comment]
The second prettier check is still failing for your formatting, could you please check the formatting of your file? i.e. the code you've added is property formatted? you can take reference from other files  in the repo. 
There is mo description added on the floats.
The branch is not updated, you need to take latest from the base.

**Comment by ethan-master-coding** [comment]
@pinalpithadiya I'm not sure what made it fail. I'm sure it will pass once all 3 reviews are done.

File: content/css/concepts/float/float.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+---
+Title: 'Floats'
+Description: ''
```
Comment (SSwiniarski) [inline content/css/concepts/float/float.md:3]:
```suggestion
Description: 'The float property moves an element to the left or right within its parent, outside of the default positioning.'
```

File: content/css/concepts/float/float.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  - 'Web Development'
+  - 'Web Design'
+Tags:
+  - 'Floats'
+CatalogContent:
+  - 'learn-css'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The `float` propertie moves an element with in its parent.
```
Comment (SSwiniarski) [inline content/css/concepts/float/float.md:14]:
```suggestion
The float property moves an element to the left or right within its parent, outside of the default positioning.
```

File: content/css/concepts/float/float.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## Syntax
+
+The `float` property takes 3 different values. They are, `left`, `right`, and `none`. The value `left` makes the element float on the left of a parent. The value `right` makes the element float on the right of a parent. The value `none` makes the element not float at all.
```
Comment (SSwiniarski) [inline content/css/concepts/float/float.md:18]:
```suggestion
The `float` property takes 3 different values. They are, `left`, `right`, and `none`. The value `left` makes the element float on the left of a parent. The value `right` makes the element float on the right of a parent.  The floating element will render beside elements with the default positioning. The value `none` removes the floating property from an element.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@ethan-master-coding, I did a first review and suggested some changes. Also, you should probably do a yarn format on your PR to fix the failing checks.

**Comment by ethan-master-coding** [comment]
All edits look good. I accepted the changes. And I'm not sure what's making the check fail.

File: content/css/concepts/grids/grids.md [hunk 5]
```diff
@@ -13,7 +13,7 @@ CatalogContent:
   - 'paths/front-end-engineer-career-path'
 ---
 
-CSS Grid is a two-dimensional grid-based layout system that uses rows, columns, and gaps to organize content on a web page. This system helps with creating a responsive layout for web pages without having to use positioning, floats, or Flexbox.
+CSS Grid is a two-dimensional grid-based layout system that uses rows, columns, and gaps to organize content on a web page. This system helps with creating a responsive layout for web pages without having to use positioning, [floats](https://www.codecademy.com/resources/docs/css/floats), or Flexbox.
```
Comment (Dusch4593) [inline content/css/concepts/grids/grids.md:16]: If we're gonna link to floats then let's also link to other terms as well:

```suggestion
CSS Grid is a two-dimensional grid-based layout system that uses rows, columns, and gaps to organize content on a web page. This system helps with creating a responsive layout for web pages without having to use [positioning](https://www.codecademy.com/resources/docs/css/position), [floats](https://www.codecademy.com/resources/docs/css/floats), or [Flexbox](https://www.codecademy.com/resources/docs/css/flexbox).
```

**Comment by Dusch4593** [comment]
> @SSwiniarski All edits look good. I accepted the changes. And I'm not sure what's making the check fail.

After pushing the changes in 39c8bf2160c4520d5d74362eb2f8fa3d54c0d33e, all the GH checks should be passing. It looked like your indentation on lines # 22 - 32 were a bit off.

Also, this is important, but always make sure that the meta-`Title` and its corresponding file + folder match. The entry is entitled "Floats" (plural) but the folder + file were named `float/` and `float.md`, respectively. (singular).

**Comment by ethan-master-coding** [comment]
@Dusch4593 All edits look good. I accepted the changes.

File: content/css/concepts/floats/floats.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+.item_1 {
+  float: left;
+}
+
+.item_2 {
+  float: right;
+}
+```
+
+The div with the class `item_1` will float to the left of the text `This example shows CSS floats.` And the div with the class `item_2` will float to the right of the text `This example shows CSS floats.`
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:44]: Can include a supporting image for this in `media/`?

Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:44]: Can include a supporting image for this in `media/`?

Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:44]:
```suggestion
The `<div>` tags with the `.item_1` and `.item_2` classes will float to the left and right of the "CSS floats" text, respectively.
```

File: content/css/concepts/floats/floats.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  <div class="item_1">
+    <p>This is the first item</p>
+  </div>
+  <div class="item_2">
+    <p>This is the second item</p>
+  </div>
+  <p>This example shows CSS floats.</p>
+</section>
+```
+
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:33]: Add 1-2 transition sentences that talk about the setup in the previous \```html snippet and how they can be selected in a separate stylesheet with class selectors (`.item_x`, etc.).

File: content/css/concepts/floats/floats.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  - 'Floats'
+CatalogContent:
+  - 'learn-css'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The float property moves an element to the left or right within its parent, outside of the default positioning.
+
+## Syntax
+
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:17]: For the "Syntax" it is preferable to include a \```pseudo code snippet that visualizes the values you're describing here.

File: content/css/concepts/floats/floats.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+The `float` property takes 3 different values. They are, `left`, `right`, and `none`. The value `left` makes the element float on the left of a parent. The value `right` makes the element float on the right of a parent. The floating element will render beside elements with the default positioning. The value `none` removes the floating property from an element.
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:18]:
```suggestion
The `float` property primarily uses the following three values:

* The `left` value that makes the element float on the left of its container.
* The `right` value that makes the element float on the right of its container.
* The `none` value that causes the element not to float.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@ethan-master-coding Left a few more comments/suggestions for you to check out.

**Comment by ethan-master-coding** [comment]
@Dusch4593 All edits look good. I accepted the changes and made the requested changes. I did add some more code just so it looks good when rendered when it's zoomed in. (How I took the screenshot.)

File: content/css/concepts/floats/floats.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  <div class="item_1">
+    <p>This is the first item and I will float left.</p>
+  </div>
+  <div class="item_2">
+    <p>This is the second item and I will float right</p>
+  </div>
+    <br>
+    <br>
+    <br>
+  <p>This example shows CSS floats and I wont float at all.</p>
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:52]:
```suggestion
<section>
<div class="item_1">
  <p>This is the first item and I will float left.</p>
</div>
<div class="item_2">
  <p>This is the second item and I will float right</p>
</div>
  <br>
  <br>
  <br>
<p>This example shows CSS floats and I wont float at all.</p>
```

File: content/css/concepts/floats/floats.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+The `float` property moves an element to the left or right within its containing element, outside of the default [positioning](https://www.codecademy.com/resources/docs/css/position).
+
+## Syntax
+
+The `float` property primarily uses the following three values:
+
+* The `left` value that makes the element float on the left of its container.
+* The `right` value that makes the element float on the right of its container.
+* The `none` value that causes the element not to float.
+
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:23]:
```suggestion
* The `none` value that causes the element not to float.
```

File: content/css/concepts/floats/floats.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+.item_2 {
+  float: right;
+}
+
+div {
+  border: thick double black;
+}
+```
+
+The `<div>` tags with the `.item_1` and `.item_2` classes will float to the left and right of the "CSS floats" text, respectively.
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:71]: Should we center the "CSS floats" text?

```css
section > p {
  text-align: center;
}
```

File: content/css/concepts/floats/floats.md (hunk 11 above)
Comment (ethan-master-coding) [reply content/css/concepts/floats/floats.md:23]: @Dusch4593 There's no change here. What is it supposed to be?

File: content/css/concepts/floats/floats.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+![Result of the above code](https://raw.githubusercontent.com/Codecademy/docs/Floats/media/css_floats.png)
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:73]:
```suggestion
![Image of rendered float elements](https://raw.githubusercontent.com/Codecademy/docs/Floats/media/css_floats.png)
```

File: content/css/concepts/floats/floats.md (hunk 11 above)
Comment (Dusch4593) [reply content/css/concepts/floats/floats.md:23]: Delete line  # 23 (the extra empty line)

File: content/css/concepts/floats/floats.md (hunk 12 above)
Comment (ethan-master-coding) [reply content/css/concepts/floats/floats.md:71]: @Dusch4593 I cant remember what the `>` sign does. Can you remind me?

Comment (Dusch4593) [reply content/css/concepts/floats/floats.md:70]: It selects all `<p>` tags where it's containing parent tag is a `<section>` element.

**Comment by ethan-master-coding** [comment]
@Dusch4593 All edits look good. I accepted the changes.

File: content/css/concepts/floats/floats.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+<div class="item_1">
+  <p>This is the first item and I will float left.</p>
+</div>
+<div class="item_2">
+  <p>This is the second item and I will float right</p>
+</div>
+  <br>
+  <br>
+  <br>
+<p>This example shows CSS floats and I wont float at all.</p>
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:51]:
```suggestion
<section>
  <div class="item_1">
    <p>This is the first item and I will float left.</p>
  </div>
  <div class="item_2">
    <p>This is the second item and I will float right</p>
  </div>
    <br>
    <br>
    <br>
  <p>This example shows CSS floats and I wont float at all.</p>
</section>
```

Noticed a missing closing `</section>`

**Comment by ethan-master-coding** [comment]
@Dusch4593 Accepted the change. Probably missed when copying and pasting between GH and replit when I was taking the screenshot.

File: content/css/concepts/floats/floats.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+  float: left;
+}
+
+.item_2 {
+  float: right;
+}
+
+section > p {
+  border: thick double black;
+}
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:68]: Actually let's keep that `<div>` rule for the borders and add the following additional rule:

```suggestion
div {
  border: thick double black;
}

section > p {
  margin: auto;
  text-align: center;
  width: 50%;
}
```

**Comment by Dusch4593** [comment]
> @Dusch4593 Accepted the change. Probably missed when copying and pasting between GH and replit when I was taking the screenshot.

Gotcha! I tend to do that myself whenever I'm transferring stuff from [Workspaces](https://www.codecademy.com/workspaces)

**Comment by ethan-master-coding** [comment]
> Gotcha! I tend to do that myself whenever I'm transferring stuff from [Workspaces](https://www.codecademy.com/workspaces)

Just saying that link brings me to a 404 page. Leecked feature on CC? 😂

**Comment by Dusch4593** [comment]
> > Gotcha! I tend to do that myself whenever I'm transferring stuff from [Workspaces](https://www.codecademy.com/workspaces)
> 
> Just saying that link brings me to a 404 page. Leecked feature on CC? 😂

Lol I don't think. It's probably just not a valid URL on their site. Workspaces is cool, though. I think you can find it when you're logged into Codecademy and you scroll down a bit. I think you can also bring up a list of _your_ Workspaces with the following URL: https://www.codecademy.com/users/your-username/workspaces

**Review by Dusch4593 — APPROVED** [review]
Things LGTM! Thank you for bearing with me for this one, @ethan-master-coding 💪

**Comment by ethan-master-coding** [comment]
> > > Gotcha! I tend to do that myself whenever I'm transferring stuff from [Workspaces](https://www.codecademy.com/workspaces)
> > 
> > 
> > Just saying that link brings me to a 404 page. Leecked feature on CC? 😂
> 
> Lol I don't think. It's probably just not a valid URL on their site. Workspaces is cool, though. I think you can find it when you're logged into Codecademy and you scroll down a bit. I think you can also bring up a list of _your_ Workspaces with the following URL: https://www.codecademy.com/users/your-username/workspaces

Oh yea here's the [link](https://www.codecademy.com/users/EthanMasterProgram/workspaces). I forget about them because I never use them.

**Comment by ethan-master-coding** [comment]
@Dusch4593 no worries. Im just waiting on the new beta for Learn Linux to go live. [Forum post](https://discuss.codecademy.com/t/introduction-to-linux-beta/660056?u=ethanmasterprogram)

File: content/css/concepts/floats/floats.md [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+section > p {
+  margin: auto;
+  text-align: center;
+  width: 50%;
+}
+```
+
+The `<div>` tags with the `.item_1` and `.item_2` classes will float to the left and right of the "CSS floats" text, respectively.
+
+![Image of rendered float elements](https://raw.githubusercontent.com/Codecademy/docs/Floats/media/css_floats.png)
```
Comment (Dusch4593) [inline content/css/concepts/floats/floats.md:79]: @ethan-master-coding Actually _super_ last comment/suggestion: Can we replace this image with a new one that reflects the style changes we just added? (with the "CSS Floats" text in the middle)

Comment (Dusch4593) [reply content/css/concepts/floats/floats.md:79]: Beautiful!

**Comment by ethan-master-coding** [comment]
@Dusch4593 Made the edit.

**Comment by Dusch4593** [comment]
@ethan-master-coding Awesome!

How do things look on your end, @SSwiniarski? Does this seem ready to merge?

**Review by SSwiniarski — APPROVED** [review]
LGTM! 👍

**Comment by ethan-master-coding** [comment]
@Dusch4593 @SSwiniarski Just took away a `.` and replaced it to force the checks to rerun so they would all run. (One of the checks were not running)

File: content/css/concepts/floats/floats.md [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+
+## Example
+
+```html
+<section>
+  <div class="item_1">
+    <p>This is the first item and I will float left.</p>
+  </div>
+  <div class="item_2">
+    <p>This is the second item and I will float right</p>
```
Comment (yangc95) [inline content/css/concepts/floats/floats.md:46]:
```suggestion
    <p>This is the second item and I will float right.</p>
```

File: content/css/concepts/floats/floats.md [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+  </div>
+  <br />
+  <br />
+  <br />
+  <p>This example shows CSS floats and I wont float at all.</p>
+</section>
+```
+
+In the above snipit the elements with the class `item_1` and `item_2` you can set the float in an external style sheet.
```
Comment (yangc95) [inline content/css/concepts/floats/floats.md:55]:
```suggestion
In the above snippet the elements with the class `item_1` and `item_2` you can set the float in an external style sheet.
```

File: content/css/concepts/grids/grids.md [hunk 19]
```diff
@@ -13,7 +13,7 @@ CatalogContent:
   - 'paths/front-end-engineer-career-path'
 ---
 
-CSS Grid is a two-dimensional grid-based layout system that uses rows, columns, and gaps to organize content on a web page. This system helps with creating a responsive layout for web pages without having to use positioning, floats, or Flexbox.
+CSS Grid is a two-dimensional grid-based layout system that uses rows, columns, and gaps to organize content on a web page. This system helps with creating a responsive layout for web pages without having to use [positioning](https://www.codecademy.com/resources/docs/css/position), [floats](https://www.codecademy.com/resources/docs/css/floats), or [Flexbox](https://www.codecademy.com/resources/docs/css/flexbox).
```
Comment (yangc95) [inline content/css/concepts/grids/grids.md:16]:
```suggestion
CSS Grid is a two-dimensional grid-based layout system that uses rows, columns, and gaps to organize content on a web page. This system helps with creating a responsive layout for web pages without having to use [positioning](https://www.codecademy.com/resources/docs/css/position), [floats](https://www.codecademy.com/resources/docs/css/floats), or [flexbox](https://www.codecademy.com/resources/docs/css/flexbox).
```

**Comment by yangc95** [comment]
@ethan-master-coding I took a look at this as well! I believe it should be ready for merge afterward 🚀

**Comment by ethan-master-coding** [comment]
@yangc95 All edits look good. I accepted the changes.

**Review by yangc95 — APPROVED** [review]
Cool @ethan-master-coding ! This LGTM

**Comment by yangc95** [comment]
@ethan-master-coding Actually! One last thing is to add a period on the image for the second item on the right 😵‍💫 it's so tedious but if you can edit that image that would be amazing

**Comment by ethan-master-coding** [comment]
@yangc95 Changed the image. Same name so I don't need to edit the image link

**Comment by yangc95** [comment]
> @yangc95 Changed the image. Same name so I don't need to edit the image link

Cool, I see that. Is there a reason the image is especially larger?

**Comment by ethan-master-coding** [comment]
@yangc95 Mabey I zoomed in a different amount. Im not sure thought.

**Comment by yangc95** [comment]
> @yangc95 Mabey I zoomed in a different amount. Im not sure thought.

Would you like to try making it smaller and with the right orientation? I can also go into your branch and directly make the changes if you'd like.

**Comment by ethan-master-coding** [comment]
@yangc95 How is this:

image

**Comment by ethan-master-coding** [comment]
Entry link: https://www.codecademy.com/resources/docs/css/floats

**Comment by yangc95** [comment]
@ethan-master-coding on top of it! ❤️ 'd the back and forth just now~

ty to @Dusch4593 @SSwiniarski 😌

**Pull request merged**